bin/reth-benchmark/src/benchmark/rpc.rs

```diff
@@ -0,0 +1,177 @@
+
+        // put results in a summary vec so they can be printed at the end
+        // TODO: just accumulate on the fly
+        let mut results = Vec::new();
+
+        while let Some(block_res) = block_stream.next().await {
+            let block = block_res?.ok_or(BlockResponseError::BlockStreamNone)?;
+            let block = match block.header.hash {
+                Some(block_hash) => {
+                    // we can reuse the hash in the response
+                    Block::try_from(block)?.seal(block_hash)
+                }
+                None => {
+                    // we don't have the hash, so let's just hash it
+                    Block::try_from(block)?.seal_slow()
+                }
+            };
+
+            // see the todo in the block fetcher for ways this can be improved
+            let head_block_hash = block.hash();
+            let safe_block_hash = block_provider
+                .get_block_by_number((block.number - 32).into(), false)
+                .await?
+                .expect("safe block exists")
+                .header
+                .hash
+                .expect("safe block has hash");
+
+            let finalized_block_hash = block_provider
+                .get_block_by_number((block.number - 64).into(), false)
+                .await?
+                .expect("finalized block exists")
+                .header
+                .hash
+                .expect("finalized block has hash");
+
+            // just put gas used here
+            let gas_used = block.header.gas_used as f64;
+
+            let versioned_hashes: Vec<B256> =
+                block.blob_versioned_hashes().into_iter().copied().collect();
+            let (payload, parent_beacon_block_root) = block_to_payload_v3(block);
+
+            debug!(
+                number=?payload.payload_inner.payload_inner.block_number,
+                hash=?payload.payload_inner.payload_inner.block_hash,
+                parent_hash=?payload.payload_inner.payload_inner.parent_hash,
+                "Sending payload",
+            );
+
+            let parent_beacon_block_root =
+                parent_beacon_block_root.expect("this is a valid v3 payload");
+            let start = Instant::now();
+            auth_provider
+                .new_payload_v3_wait(payload, versioned_hashes, parent_beacon_block_root)
+                .await?;
+            let new_payload_duration = start.elapsed();
+
+            // construct fcu to call
+            let forkchoice_state =
+                ForkchoiceState { head_block_hash, safe_block_hash, finalized_block_hash };
+
+            // TODO: allow the user to configure what to do when they get the output of the stream,
+            // ie, the block inside of this while loop should be configurable, along with the stream
+            // itself
+            auth_provider.fork_choice_updated_v3_wait(forkchoice_state, None).await?;
+            let fcu_duration = start.elapsed() - new_payload_duration;
```

Comment:
cc @mattsse I suspect the newPayload -> FCU codepath is the one that will be followed in any protocol that doesn't have multiple forks near the tip e.g. with the BFT integration?